### PR TITLE
Epinephrine buff

### PR DESCRIPTION
**Pull request by Arendian**
## About the PR

Changed the working condition of the chem from being in a critical state to having more than 90 damage.
Added pressure immunity while above 90 damage.
Doubled the brute and burn healing from 0.5/u to 1/u(heals only above 90 damage)

## Why / Balance

There is basically no reason currently to make epinephrine, because it's outclassed by other chemicals that are way easier to make.
This change will give it a niche of allowing you to stay at 90 damage, even in spaced areas, as long as no additional damage of any kind is dealt.

Emergency pens contain 12u of epinephrine, which before this PR healed 6 brute and 6 burn as long as the mob was crit, now one emergency pen heals 12 brute and 12 burn as long as the person is above 90 damage.
One emergency pen will allow someone to walk in a spaced area for 24 seconds(only above 90 damage), but they will be extremely slowed down because of the damage slowdown. So it's not just a reintroduction of the spacepen, which allowed people to pass spaced areas with minimal harm and no slowdown if used before entering the spaced area.
Epinephrine making people pressure immune above 90 damage also makes it so someone in a hardsuit has a bit more time to reach an unspaced area when trying to save someone without a hardsuit.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Dygon
- tweak: Epinephrine now heals while above 90 damage, has it's brute and burn healing doubled, and makes people above 90 damage pressure immune.

### CODE REVIEW

**Comment by Aisu9** [comment]
space cold damage will most likely still kill them

**Comment by Arendian** [comment]
> space cold damage will most likely still kill them

This is intended, you only cool down enough for it to become an issue if you spend 10+ seconds in space or on a lattice, which most of the time means you're not inside the station anymore. If you just want to reach the closest unspaced area while inside the station you will probably have a tile under your feet.

**Comment by Aisu9** [comment]
what about lizards? they will start to take more cold damage much faster than anyone else

**Comment by Arendian** [comment]
> what about lizards? they will start to take more cold damage much faster than anyone else

Lizards still won't get cold as long as there is a tile under their feet.

**Comment by nikthechampiongr** [comment]
This seems like a roundabout way to revert #23970 by combining the space pen's affect with epinephrine.

**Comment by nikthechampiongr** [comment]
It also significantly buffs epinephrine even without the pressure immunity. Epinephrine is one of the most used chems in the game and is extremely useful with dealing with people in critical condition. Chemist regularly make it although it's uncommon compared to the usual chems lik bica/derma/dex+. The reason chemists don't produce it more often is because the recipe is relatively hard, and everyone already starts with an emergency pen.

**Comment by Krunklehorn** [comment]
Epinephrine is just a crit stablizer. It's not supposed to be fantastic at healing, it's supposed to give you time.

The threshold change is cool, but only because repeatedly standing up and falling over is annoying. It doesn't justify adding barotrauma immunity.

A full health player will be able to charge through a spaced area just like they used to with barozine. Moving the effect to epinephrine would allow players to manufacture it? That sounds worse.

**Comment by Arendian** [comment]
> It also significantly buffs epinephrine even without the pressure immunity. Epinephrine is one of the most used chems in the game and is extremely useful with dealing with people in critical condition. Chemist regularly make it although it's uncommon compared to the usual chems like bica/derma/dex+. The reason chemists don't produce it more often is because the recipe is relatively hard, and everyone already starts with an emergency pen.

The only reason epinephrine is being used is because it's in the emergency medipen. I also don't agree with your statement about it  being "regularly made" I haven't seen it made a single time since the limited chems, and before that I think I can count the amount of rounds I've seen it made(and used) on two hands.

Inaprovaline is a better stabilizer because it stops bleeding and heals more oxygen damage than epinephrine(it's also way easier and cheaper to make).
Dex+ is a bit harder to make than inaprovaline(only because of the plasma) but still way easier than epinephrine. It heals more oxygen damage than epinephrine and additionally heals tons of bloodloss damage.

The only thing what epinephrine does what the above mentioned chemicals don't is give you minor stun reduction(not useful for a stabilizer) and a tiny amount of brute/burn/poison healing. 10u of of epinephrine heals 3.3 blunt(1.66u bicaridine) damage or 2.5 heat(1.66u dermaline/7.5u kelotane), and only when someone is crit.

I don't see why anyone would currently use/make epinephrine(besides RP reasons maybe) if it doesn't come in a medipen that allows you to instant inject it.


> This seems like a roundabout way to revert #23970 by combining the space pen's affect with epinephrine.

Space pens were something you used before you went into a spaced area, and kept you in peak fighting condition until it stopped working. The change suggested in this PR gives you extra time to reach an unspaced area while heavily wounded and barely hanging on to your life. In my opinion these are two different things, but if too many people think the same as you I think limiting the pressure immunity to when someone is crit could be a compromise.

> Epinephrine is just a crit stablizer. It's not supposed to be fantastic at healing, it's supposed to give you time.

Inaprovaline takes way less effort to make and it even heals more airloss, the only thing epinephrine adds is stun resistance(you don't need this when crit), and a negligible amount of healing. Wihout this PR 12u of epinephrine(emergency medipen) won't even heal a punch worth of damage.

> A full health player will be able to charge through a spaced area just like they used to with barozine. Moving the effect to epinephrine would allow players to manufacture it? That sounds worse.

They won't stay full health, the healing and pressure immunity only kick in when someone has more than 90 damage.

File: Resources/Prototypes/Reagents/medicine.yml [hunk 1]
```diff
@@ -351,6 +350,12 @@
           min: 1
         reagent: Histamine
         amount: 4
+      - !type:GenericStatusEffect
+        conditions:
+        - !type:TotalDamage
+          min: 90
+        key: PressureImmunity
+        component: PressureImmunity
```
Comment (VasilisThePikachu) [inline Resources/Prototypes/Reagents/medicine.yml:358]: I feel like pure immunity is too much. even if you are at 90hp.

I can imagine players now saving the medipen for themselves in case they get in a spaced environment instead of helping others. Along with that since epi can be made. You can mass produce a lot and keep injecting yourself for as much as you have epi on a jug.

This is just a suggestion. I'm a contributor so you are free to ignore this.

**Comment by Rainbeon** [comment]
> Chemist regularly make it

Epi is a Chemist vanity project and not actually useful in a Medbay setting over Dex+/Inaprov, similar to Omni, weak healing Reagents that are more forgiving to be blind-injected are only really useful outside of the Medbay (like in Emergency Medipens)

**Comment by UbaserB** [comment]
> > Chemist regularly make it
> 
> Epi is a Chemist vanity project and not actually useful in a Medbay setting over Dex+/Inaprov, similar to Omni, weak healing Reagents that are more forgiving to be blind-injected are only really useful outside of the Medbay (like in Emergency Medipens)
> 
> That being said, it's good at what it does do, and I don't really think it needs buffs (and that Chemists should just stop wasting their time making this instead)

No, its actually really powerful lol. You doctors dont notice, and it's the reason why i always try to make epi as a chemist.

**Comment by Jezithyr** [comment]
Can we not turn epi into spacepen 2? Spacepen/pressure immunity was removed for a specific reason. Epi is already very strong, buffing it further is not really neccessary IMO.

**Comment by Holinka4ever** [comment]
> Can we not turn epi into spacepen 2? Spacepen/pressure immunity was removed for a specific reason. Epi is already very strong, buffing it further is not really neccessary IMO.

People already told that epi is not such usefull right now due of complex craft after chemistry refractor.
And it doesnt look like space pen due of 90 damage treshold, you still die from cold damage wile having 10 hp with perma cold movement. Truely will be funny, if pepople will try to use it as powergame weapon to beat nukies, yeah.
It mostly will used to heal people in critical state.

**Comment by Ilya246** [comment]
i don't see how this is spacepen 2 considering:
1) it does not save you against the cold, so you will still die without a way to either negate or heal cold
2) it only works above 90 hp, so you will be heavily slowed down and unable to use this in combat
3) it's basically a timed emergency suit in the form of a pen

**Comment by K-Dynamic** [comment]
Cold dmg isnt usually the killer, it's the 2.4 blunt dmg/s in a vacuum environment

If pressure immunity only allowed above 90 dmg then it's a very weak space pen and last-ditch escape when you're moving at like 10% speed, which doesn't seem that bad considering how unforgiving firelocks can be

It'll also be good for stabilising crit people in spaced areas and dragging them rather than people leaving them to die

**Comment by Jezithyr** [comment]
Pressure immunity was removed for a reason and it should not be re-added. It made surviving spacing too easy and is not healthy for the game. The other changes are fine, but pressure immunity is a no-go regardless of whether it takes effect at a particular threshold. Spacing is SUPPOSED to be dangerous.

**Comment by Jezithyr** [comment]
> i don't see how this is spacepen 2 considering:
> 
>     1. it does not save you against the cold, so you will still die without a way to either negate or heal cold
> 
>     2. it only works above 90 dmg, so you will be heavily slowed down and unable to use this in combat
> 
>     3. it's basically a timed emergency suit in the form of a pen _but_ only above 90 dmg
> 
> 
> i must ask what problems do the opponents of this think it will introduce and why are they bad, "it's spacepen 2" isn't really an argument

Pressure immunity is the problem, spacing and lack of pressure are supposed to BE dangerous. Allowing epi (which is pretty common) to negate the effects of pressure loss, even if it requires a specific health threshold reduces the danger of spacing. It's powercreep, especially after spacepen was removed

**Comment by Ilya246** [comment]
> Pressure immunity is the problem, spacing and lack of pressure are supposed to BE dangerous. Allowing epi (which is pretty common) to negate the effects of pressure loss, even if it requires a specific health threshold reduces the danger of spacing. It's powercreep, especially after spacepen was removed.
> 
> TLDR: It's not exactly spacepen2 but it's a ways down the road to getting there and that's why there is an issue. It's power creep plain and simple.

"powercreep" doesn't really mean or prove anything (you could call any buff or beneficial feature powercreep), and as for it making spacing survivable, i think it's good as it allows for more fun to be had from clutches; i don't see why you want spacing to be this unavoidable (and not even delayable) death thing unless you have an EVA
note that once you start taking cold damage, you might as well count yourself dead, because you will likely die even if you get inside an atmos, as your body temp would take time to rise; basically, you can't use it to spacewalk
then, again, we already allow people to survive spacing (in an even better way) by simply finding an energency suit
however, what i would propose is to also make epi react with derma and/or leporazine to make bootleg spacewalks harder

**Comment by Holinka4ever** [comment]
> Pressure immunity is the problem, spacing and lack of pressure are supposed to BE dangerous. 

Its still dangerous, epi pen will not save you from this danger, you just giving you a chance to live.


> Allowing epi (which is pretty common) to negate the effects of pressure loss, even if it requires a specific health threshold reduces the danger of spacing.

After deleting infinity chemistry, making epi become less valuable toward to inaprovaline.
Yeah it also heal a damage, but healing group damage is bad on statistics.

> It's powercreep, especially after spacepen was removed.

Its just a lovely buff to epi, you told that its asking for specifec health treshold that is 90 damage and above to WORK.
Just clear question, what you gonna do with 24 seconds of spaceproof, thats not negating cold damage with huge walk debuff, making you valid target for one hit of toolbox?

> TLDR: It's not exactly spacepen2 but it's a ways down the road to getting there and that's why there is an issue. It's power creep plain and simple.

We lost meaning of powercreeping,  sometimes is just a hidden buff to use, sometimes just qol buff.
We have a lot of prs that merged in left and right, that giving significiant power boost. Like soap reagent that got huge stun, acidificier using as weapon to hide bodies quickly, knifes to sec.
A lot of things really. But giving epi spaceproof that ONLY working in special life threshould cant be call powercreep due of that.

P.S. If you want to argument my subjective opinion, find a pep whom gonna use epi as space pen, it would be funny to see pep whom using chemicals instead of eva suits.

**Comment by deltanedas** [comment]
the big spacepen thing was it kept you at close to 0 damage for a while, which made it great for validhunting anything in space. you wont be able to validhunt anything at 90 damage

**Comment by Jezithyr** [comment]
> Its still dangerous, epi pen will not save you from this danger, you just giving you a chance to live.

This is literally the definition of powercreep. Pressure immunity was previously provided by spacepen and that was removed. Spacepen did not start out being as overtuned as it ended up, it started from smaller changes like this. AKA the "Creep" in power creep. Many small unhealthy changes that overtime end up creating something bad for the health of the game.

> Its just a lovely buff to epi, you told that its asking for specifec health treshold that is 90 damage and above to WORK.
Just clear question, what you gonna do with 24 seconds of spaceproof, thats not negating cold damage with huge walk debuff, making you valid target for one hit of toolbox?
Epi is meant to be a drug that is used to stabilize someone close to death, NOT as something you inject to save yourself from being spaced. The pressure immunity is not needed to achieve this and IS power creep, don't even try to sugarcoat it. 

As it stands right now, this PR will not be merged with the pressure immunity change regardless of if it has a health threshold or not.

File: Resources/Prototypes/Reagents/medicine.yml [hunk 2]
```diff
@@ -351,6 +350,12 @@
           min: 1
         reagent: Histamine
         amount: 4
+      - !type:GenericStatusEffect
+        conditions:
+        - !type:TotalDamage
+          min: 90
+        key: PressureImmunity
```
Comment (Jezithyr) [inline Resources/Prototypes/Reagents/medicine.yml:357]: Absolutely not. No pressure immunity, not even with a health threshold. Unless this is removed, this PR will not be merged.

**Review by Jezithyr — CHANGES_REQUESTED** [review]

**Comment by Holinka4ever** [comment]
> > Its still dangerous, epi pen will not save you from this danger, you just giving you a chance to live.
> 
> This is literally the definition of powercreep. Pressure immunity was previously provided by spacepen and that was removed.

Because people abused it, as deltanedas said.
Cause in place were i played, people used space pens as weapons to deal a lot of damage, as a botleg space suit, or option to escape death.

> Spacepen did not start out being as overtuned as it ended up, it started from smaller changes like this.

Sorry for next sentence, but thats kinda weird that someone allowed to that creep to grow stronger each bit.

> AKA the "Creep" in power creep. Many small unhealthy changes that overtime end up creating something bad for the health of the game.

Yeah, i know about enough, that thing can really frustrate players at some point.

> Epi is meant to be a drug that is used to stabilize someone close to death, NOT as something you inject to save yourself from being spaced. 

Yeah,  maybe i was a bit of rush for epi buff, cause, people still use it because it got buffed, nor got a space immunity

>The pressure immunity is not needed to achieve this and IS power creep, don't even try to sugarcoat it.

You right, even with debuffs, someone would just make a prs to buff these treshold for own comfort zone.

> As it stands right now, this PR will not be merged with the pressure immunity change regardless of if it has a health threshold or not. If you do not want to make this change, then you can close the PR or say that you are unwilling and I will do it for you.

Ohh, by the way, i am just comentator that rushed for epi buff, nor a contributor, so at this point... what about to make epi heal more then, as omnizine and more, but in critical state?

P.S. 
Yeah we see that space immunity would become long term powercreeping blister, because people can use it for own profit... what if about to return spacepen... that working **only** in critical state?
Yes it sounds weird, but just a thought, nor important one, but just exists

File: Resources/Prototypes/Reagents/medicine.yml (hunk 2 above)
Comment (Arendian) [reply Resources/Prototypes/Reagents/medicine.yml:357]: Are you also against the space immunity if the threshold is the person being crit? This way people can't use it on themselves, but other people can keep someone stable for a bit longer on the way out of a spaced area.

File: Resources/Prototypes/Reagents/medicine.yml (hunk 1 above)
Comment (Arendian) [reply Resources/Prototypes/Reagents/medicine.yml:358]: Do you think it only making you pressure immune while in crit would be balanced? You would need someone else to reapply the epi to you, and you won't be able to do anything untill they put a suit on you or drag you into an unspaced area.

**Comment by Jezithyr** [comment]
> Sorry for next sentence, but thats kinda weird that someone allowed to that creep to grow stronger each bit.

Yeah, a lot of the spacepen changes occurred before we had standardized design documentation and guidelines for reviewing design PRs. (Plus I was busy with work so I wasn't reviewing PRs)

> Yeah we see that space immunity would become long term powercreeping blister, because people can use it for own profit... what if about to return spacepen... that working only in critical state?
There isn't a need for pressure immunity while you are in crit. In crit you can't move/interact, so you are entirely at the mercy of someone else saving you. Simply giving you a bunch of health to stabilize you will give whoever is rescuing you more than enough time to get you out of the spaced area without needing to provide immunity. Pressure immunity being given by a drug doesn't even make much sense anyway.

**Comment by Jezithyr** [comment]
The problem with adding pressure immunity as a regular drug effect is that it opens pandora's box in terms of power-creep issues. Plus it doesn't really fit the Epi's intended role, which is basically a crit stabilizer. 
There are also behavioural-design implications of giving pressure immunity to a stabilizer drug. The intended gameplay for epi is for it to be used to stabilize someone in crit so that you have enough time to remove them from a dangerous situation so that you can heal them or focus on other injured people. 

Being exposed to vacuum is a dangerous situation and if epi provides pressure immunity (even in crit), there is no longer an incentive for a player to move the injured person into a pressurized area. 
This is a problem as it effectively trivializes responding to spacing for parameds/doctors, since they can just epi everyone and take their time with treatment, whereas currently they need to prioritize getting their patients to a pressurized area.

**Comment by Holinka4ever** [comment]
Seems, space immunity out from solution to make epi good, cause or it will be broken, or nor concept well-written into.
I actually got mandela effect from livrah streams, that in tg station epi is also used to hold bodies from being rot, but actually its epipen itself, that have formaldehyde inside.
So what about to give rot prevent instead of space immunity?
It would safe many lifes begin with, and it cant be used due of combat or related stuff.
So, someone would say that you can use body bag, morgue and freeze box. Yes you right! And having more options to hold bodies from rot is good for diversity.
Also not a lot of people having body bag, but they do have epi pen. Thats my thougt.

**Comment by deltanedas** [comment]
>  there is no longer an incentive for a player to move the injured person into a pressurized area to stabilize them.

there very much still is:
1. patient is 150 damage in spaced area
2. inject epi, they are now stable at 100 damage and not taking pressure damage
3. you start healing
4. oops its still spaced they immediately go crit again

having space protection when crit (or just high brute healing) would mean they wouldnt die while being moved out of a spaced area

**Comment by Jezithyr** [comment]
> having space protection when crit (or just high brute healing) would mean they wouldnt die while being moved out of a spaced area

Like you said, they wouldn't die being moved out of the spaced area. With pressure immunity there is no risk to just give them epi and then coming back later to bring them to a safe pressure area. Which is a behaviour that we don't want, medics should be incentivized to prioritize moving patients out of a spaced area as fast as possible not stabilize them at 90hp insitu and move on because the patient is pressure immune.

**Comment by Jezithyr** [comment]
> Seems, space immunity out from solution to make epi good, cause or it will be broken, or nor concept well-written into. I actually got mandela effect from livrah streams, that in tg station epi is also used to hold bodies from being rot, but actually its epipen itself, that have formaldehyde inside. So what about to give rot prevent instead of space immunity? It would safe many lifes begin with, and it cant be used due of combat or related stuff. So, someone would say that you can use body bag, morgue and freeze box. Yes you right! And having more options to hold bodies from rot is good for diversity. Also not a lot of people having body bag, but they do have epi pen. Thats my thougt.

This seems like a much better solution IMO, making epi hold bodies from rotting for a short period would definitely give it more utility and give players another option for combating rot. It also fits the design intention of epi-pens being used to quickly stabilize/prevent a patient from getting worse.

**Comment by deltanedas** [comment]
> what's to stop the player from injecting the patient from epi then moving on to deal with another patient?

if you do that you will still have to treat the first patient at some point so either:
- you waste 100s of units of epi to keep them in purgatory
- you stop being a shitter and actually heal them

**Comment by Ilya246** [comment]
what if epi metabolised more slowly but increased the death threshold from 200 to like 250

**Comment by Jezithyr** [comment]
> > what's to stop the player from injecting the patient from epi then moving on to deal with another patient?
> 
> if you do that you will still have to treat the first patient at some point so either:
> 
>     * you waste 100s of units of epi to keep them in purgatory
> 
>     * you stop being a shitter and actually heal them

Pressure immunity still provides a competing incentive for multi-person spacing situations, so it's not really a good solution from a design perspective. Expecting a player to "Stop being a shitter" is not good design :P

**Comment by metalgearsloth** [comment]
This got turbo bikeshedded to hell but looks like it's DNM til newmed or something anyway.
